packages/embed-services/src/ThemeService.ts

```diff
@@ -0,0 +1,147 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import type { ITheme } from '@looker/sdk'
+import {
+  all_themes,
+  default_theme,
+  delete_theme,
+  theme,
+  update_theme,
+} from '@looker/sdk'
+import type { IAPIMethods } from '@looker/sdk-rtl'
+import { ItemList } from './ItemList'
+import type { IEntityService, IItemList, GetOptions } from './ItemList'
+import type { ServiceCreatorFunc } from './ServiceFactory'
+import { getFactory } from './ServiceFactory'
+
+export interface IThemeService
+  extends IItemList<ITheme>,
+    IEntityService<ITheme> {
+  defaultTheme?: ITheme
+  getDefaultTheme(ts?: Date): Promise<ITheme>
+  load(): Promise<IThemeService>
+}
+
+class ThemeService extends ItemList<ITheme> implements IThemeService {
+  public defaultTheme?: ITheme
+
+  /**
+   * Get theme by id
+   * @param id of theme to retrieve
+   * @param options to get
+   */
+  async get(id: string, options?: GetOptions): Promise<Partial<ITheme>> {
+    // TODO: implement logic to check if requested fields are already cached.
+    const cache = this.getCacheDefault(options)
+    this.clearIfExpired()
+    let item = this.indexedItems[id]
+
+    if (cache && item) {
+      return item
+    }
+
+    item = await this.sdk.ok(theme(this.sdk, id, options?.fields))
+
+    if (item) {
+      this.items = [...this.items, item]
+      this.index()
+      this.setExpiration()
+    }
+    return item
+  }
+
+  /**
+   * Get all themes
+   * @param fields to retrieve
+   */
+  async getAll(fields?: string) {
```

Comment:
this may need to be an `options?: GetOptions` param for interface consistency as we add other services with `all` calls that have different options. Doesn't need to happen now. Still need to think about smart ways to extend that interface for code hints.